src/libsystemd-network/sd-dhcp6-lease.c

```diff
@@ -473,6 +485,68 @@ int sd_dhcp6_lease_get_captive_portal(sd_dhcp6_lease *lease, const char **ret) {
         return 0;
 }
 
+int dhcp6_lease_add_vendor_sub_option(sd_dhcp6_lease *lease, const uint8_t *optval, size_t optlen) {
+        int r;
+
+        assert(lease);
+        assert(optval || optlen == 0);
+
+        for (size_t offset = 0; offset < optlen;) {
+                const uint8_t *subval;
+                size_t sublen;
+                uint16_t subopt;
+
+                r = dhcp6_parse_vendor_sub_option(optval, optlen, &offset, &subopt, &sublen, &subval);
+                if (r < 0)
+                        return r;
+
+                switch (subopt) {
+                        case DHCP6_VENDOR_SUBOPTION_BASE ... DHCP6_VENDOR_SUBOPTION_LAST:
+                                r = dhcp6_lease_insert_vendor_sub_option(lease, subopt, subval, sublen);
+                                if (r < 0)
+                                        return r;
+
+                        break;
+                }
+        }
+        return 0;
+}
+
+int dhcp6_lease_insert_vendor_sub_option(sd_dhcp6_lease *lease, uint16_t option_code, const void *data, size_t len) {
+        _cleanup_(dhcp6_vendor_sub_option_freep) DHCP6VendorOption *option = NULL;
+        DHCP6VendorOption *before = NULL;
+
+        assert(lease);
+
+        LIST_FOREACH(options, cur, lease->vendor_options) {
```

Comment:
Please do not sort here, but simply manage the options with `Set` or `OrderedSet`, then sort the vendor options later.

Reply:
Won't it be simpler to sort and manage using a list as the options are in type/length/value format. Correct me if I am wrong.

Reply:
Even though, please do not introduce in house sorting logic.
Please use `set_put()` or friends, and call `set_dump_sorted()` in a getter.

Reply:
Used set_put and set_dump_sorted()